### DISCUSSION TITLE
Deleting user not possible: Failed to delete record. Make sure that the record is not part of a required relation reference.

### DISCUSSION THREAD

**Discussion by Weaselfox** [seriousm4x/UpSnap#411] [Help wanted] [answered]
Hi, 
I have a UpSnap installation with now 16 users and 20 PCs. Now The user no. 16 was intend to use as a testuser, so after finishing the tests I tried to delete this user. Unfortunately if I confirm the question to delete the user, I get the message "Failed to delete record. Make sure that the record is not part of a required relation reference.". The user has no permissions assigned anymore and I don't know which relations could block the removal.

This message appears in the front end (/users/) as in the backend (/_/) page...

Do you have any idea, how to analyse the problem and - more important - to get rid of it?
Thanks so far...

**Comment by seriousm4x** [comment]
Hm, I've just tried to replicate this, but couldn't. What I tried:

- Create new user
- Assign permission to create new devices
- Log out and back in as the new user
- Create a new device
- Log out again and back in as the admin
- Delete user

Have you found any consistent steps to replicate this?

The only other relation that users have besides permissions is the `created_by` field on a device. Maybe there is something weird going on there?

If the user you want to delete is assigned to any `created_by` field, try to remove the user from the device and then delete the user. Lets see if that works.

**Comment by Weaselfox** [comment]
Hi, thanks so far for your message. I also tried to replicate the behaviour on an clean new install with several users, but I also couldn't. 
I've just checked the create_by column, it's for all devices N/A

What I could do to replicate is a backup of the database. If I backup the db and restore it on a clean install, the same behaviour occurs. Is it usefull for you to get the db from me?

**Reply by seriousm4x** [reply]
Ok, then I'm out of ideas.

Yes, you can send me the db. You can either send me an email to maxi@quoss.org or upload the db here if you censor all your private data from the db (like mac addresses, emails, shutdown_cmd, wake_cmd commands, ...) Then I can take a look at it.

**Reply by seriousm4x** [reply, marked as answer]
Thanks for the database. I have found the problem. It was not possible to delete users if they had a relation in the `permissions` table.
The frontend first tried to delete the user and then the permissions. I have now changed this and it should now work in version [4.2.4](https://github.com/seriousm4x/UpSnap/releases/tag/4.2.4)
